### PR TITLE
DISCO_F746NG: TDB External KVStore with QSPIF

### PR DESCRIPTION
**Pull request by ristohuhtala**
Add config for DISCO_F746NG where storage type is TDB_EXTERNAL

### CODE REVIEW

File: configs/tdb_external_kvstore_with_qspif.json [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    "|                          |",
+                    "|   Firmware Candidate 0   |",
+                    "|                          |",
+                    "+--------------------------+",
+                    "|   Firmware Candidate 0   |",
+                    "|     Metadata Header      |",
+                    "+--------------------------+ <-+ update-client.storage-address (1024 * 1024)",
+                    "|                          |",
+                    "|     External KVStore     |",
+                    "|      Size: 1MB           |",
```
Comment (teetak01) [inline configs/tdb_external_kvstore_with_qspif.json:76]: cosmetic, but this image has old numbers, should be 128 KiB

File: configs/tdb_external_kvstore_with_qspif.json [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        "*": {
+            "target.console-uart"                       : false,
+            "target.c_lib"                              : "small",
+            "platform.use-mpu"                          : false,
+            "platform.default-serial-baud-rate"         : 115200,
+            "platform.stdio-flush-at-exit"              : false,
+            "update-client.firmware-header-version"     : "2",
+            "target.restrict_size"                      : "0x8000",
+            "kvstore-size"                              : "2*16*1024",
+            "mbed-bootloader.bootloader-size"           : "(64*1024)",
```
Comment (teetak01) [inline configs/tdb_external_kvstore_with_qspif.json:102]: this should be 32*1024, and then for F746NG specifically override to 64 KiB ?
Same thing for the restrict_size calculation.

Comment (teetak01) [reply configs/tdb_external_kvstore_with_qspif.json:102]: Most other boards will probably fit the bootloader to 32 Kib ?

Comment (teetak01) [reply configs/tdb_external_kvstore_with_qspif.json:102]: actually the restrict_size is already for 32 KiB size

File: configs/tdb_external_kvstore_with_qspif.json (hunk 1 above)
Comment (ristohuhtala) [reply configs/tdb_external_kvstore_with_qspif.json]: Fixed

File: configs/tdb_external_kvstore_with_qspif.json (hunk 2 above)
Comment (ristohuhtala) [reply configs/tdb_external_kvstore_with_qspif.json]: Fixed

Comment (ristohuhtala) [reply configs/tdb_external_kvstore_with_qspif.json]: If restrict_size not defined, mbed compile uses 32KiB value

**Comment by ristohuhtala** [comment]
@teetak01 now this would ready to merge.

**Comment by ristohuhtala** [comment]
@SeppoTakalo can you press merge? I am not authorized to merge

**Pull request merged**